pynlpl/local/text/stanza_api.py

```diff
@@ -0,0 +1,64 @@
+
+class Stanza_English_default():
+  """Implements an abstracted version of the Stanza NLP library in English.
+
+    Attributes
+    ----------
+    text: str
+        The string object we want to analyze with Stanza NLP
+
+    stanza_pipeline: str
+        The name of the Stanza language model we want to use
+
+    Methods
+    ----------
+    token_segmentation(text = None)
+        split by tokens
+
+    sentence_segmentation(text = None)
+        no tokenization, text data is split by sentences only
+    """
+  def __init__(self, text: str = None, stanza_pipeline: str = 'en'):
+    '''Initializes an English Stanza NLP language model, creates a document data object if text is passed'''
+    import stanza
+    stanza.download(lang='en')
+    self.nlp = stanza.Pipeline(stanza_pipeline)
+    if text:
+      self.document = self.nlp(text)
+
+  def __validate_text(self, text):
+        if text:
+            self.document = self.nlp(text)
+            return
+        elif not self.document:
+            raise Exception("No text found. Please provide a string.")
+
+
+  def token_segmentation(self, text = None):
+    '''segments by tokens. returns list of strings'''
+    self.__validate_text(text)
+    return[token.text for token in self.document.iter_tokens()]
+
+
+  def sentence_segmentation(self, text = None):
+    '''segments by sentences only. returns list of strings'''
+    self.__validate_text(text)
+    return[sentence.text for sentence in self.document.sentences]
+
+  def pos_tagging(self, text = None):
+    '''returns list of words and their part of speech tags'''
+    self.__validate_text(text)
+    return[f"WORD: {word.text}, UPOS: {word.upos}, XPOS: {word.xpos} " for word in self.document.iter_words()]
+
+  def text_lemmatization(self, text = None):
+    '''returns list of root words for text data'''
+    self.__validate_text(text)
+    return [
+        f'WORD: {word.text}, LEMMA: {word.lemma} ' for sent in self.document.sentences
+        for word in sent.words
+    ]
+
+  def dependency_parsing(self, text = None):
```

Comment:
Find a way to explain what this does in English to someone who hasn't read the documentation

Reply:
I added some lines to the function explanation. Is this what you meant?